### PR TITLE
bpo-32746: Fix multiple typos

### PR DESCRIPTION
**Pull request by come-maiz**
Fix typos found by codespell in docs, docstrings, and comments.

https://bugs.python.org/issue32746

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/unittest/test/test_discovery.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def loadTestsFromModule(module, pattern=None):
                          ['a_directory', 'test_directory', 'test_directory2'])
 
         # load_tests should have been called once with loader, tests and pattern
-        # (but there are no tests in our stub module itself, so thats [] at the
+        # (but there are no tests in our stub module itself, so that's [] at the
```
Comment (terryjreedy) [inline Lib/unittest/test/test_discovery.py]: 'that is' would maybe be even better

**Review by terryjreedy — COMMENTED** [review]
All the changes are correct.  Perhaps "that is" in 2 places would be even better than "that's".  So 'Approved' once this change is considered.

File: Lib/test/decimaltestdata/abs.decTest [hunk 2]
```diff
@@ -20,7 +20,7 @@
 version: 2.59
 
 -- This set of tests primarily tests the existence of the operator.
--- Additon, subtraction, rounding, and more overflows are tested
```
Comment (skrah) [inline Lib/test/decimaltestdata/abs.decTest]: This is an external source.  I would prefer not to maintain our own version, even for typos.

File: Modules/_ctypes/libffi_msvc/win32.c [hunk 3]
```diff
@@ -2,21 +2,21 @@
    win32.S - Copyright (c) 1996, 1998, 2001, 2002  Red Hat, Inc.
 	     Copyright (c) 2001  John Beniton
 	     Copyright (c) 2002  Ranjit Mathew
-
```
Comment (ericvsmith) [inline Modules/_ctypes/libffi_msvc/win32.c]: I realize we have our own copy of libffi for Windows. Wouldn't changes to these files cause porting upstream changes more difficult? I suggest we not change any libffi files. See Modules/_ctypes/libffi_osx/README.ctypes.

File: Modules/_ctypes/libffi_osx/powerpc/ppc-ffi_darwin.c [hunk 4]
```diff
@@ -650,7 +650,7 @@ ffi_call(
 		  +---------------------------------------+ 160
 		  | result area 8                         |
 		  +---------------------------------------+ 168
-		  | alignement to the next multiple of 16 |
```
Comment (ericvsmith) [inline Modules/_ctypes/libffi_osx/powerpc/ppc-ffi_darwin.c]: Same comment with this file:

I realize we have our own copy of libffi for OS-X. Wouldn't changes to these files cause porting upstream changes more difficult? I suggest we not change any libffi files. See Modules/_ctypes/libffi_osx/README.pyobjc.

**Review by ericvsmith — DISMISSED** [review]
See _ctypes/libffi* comments. At this point we might not be forward or back porting any changes from upstream, but if someone wants to eventually do that or to use a new version, any extra diffs are going to make the task more difficult.

File: Lib/unittest/test/test_discovery.py (hunk 1 above)
Comment (come-maiz) [reply Lib/unittest/test/test_discovery.py]: done.

File: Lib/test/decimaltestdata/abs.decTest (hunk 2 above)
Comment (come-maiz) [reply Lib/test/decimaltestdata/abs.decTest]: reverted.

File: Modules/_ctypes/libffi_msvc/win32.c (hunk 3 above)
Comment (come-maiz) [reply Modules/_ctypes/libffi_msvc/win32.c]: reverted.

File: Modules/_ctypes/libffi_osx/powerpc/ppc-ffi_darwin.c (hunk 4 above)
Comment (come-maiz) [reply Modules/_ctypes/libffi_osx/powerpc/ppc-ffi_darwin.c]: reverted.

File: Doc/library/ossaudiodev.rst [hunk 5]
```diff
@@ -447,4 +447,3 @@ The remaining methods are specific to audio mixing:
    microphone input::
 
       mixer.setrecsrc (1 << ossaudiodev.SOUND_MIXER_MIC)
-
```
Comment (srinivasreddy) [inline Doc/library/ossaudiodev.rst:450]: Pls do not remove existing lines. In other places too

Comment (come-maiz) [reply Doc/library/ossaudiodev.rst:450]: I can revert this, but I'm curious, why do you want two empty lines at the bottom?

Comment (srinivasreddy) [reply Doc/library/ossaudiodev.rst:450]: I am sorry. I didn't look at the file. If it is at bottom, you can keep it. I am worried about introducing unnecessary git diffs in the middle of a file.

**Review by srinivasreddy — APPROVED** [review]
LGTM 👍

**Review by phoenix24 — APPROVED** [review]

**Comment by terryjreedy** [comment]
Leo, it might have been better to open a bpo issue for changes to this many files, but I don't think it necessary now, so I am adding the skip labels now.  I found your bpo user listing, and the Knight seems to agree as it did not add the CLA label back.

The conflict in aclocal.m4 appeared to be because someone added, since this patch, 
```
<blank line>
m4_include([m4/ax_check_openssl.m4])
```
at the bottom of the file, which conflicted with your removal of an extra blank line at the bottom without the context of the new line.  In any case, I removed the context markers and left the addition alone.  This should be rechecked after the retesting.

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by terryjreedy** [comment]
A few changes have been reverted.  Does anything think anything else should be?

**Comment by terryjreedy** [comment]
I decided a) to open an issue and b) to make this strictly about typos.  After previous changes, there was only one whitespace deletion left (\n at the end of a data file).

File: Lib/opcode.py [hunk 6]
```diff
@@ -142,7 +142,7 @@ def jabs_op(name, op):
 def_op('BUILD_TUPLE', 102)      # Number of tuple items
 def_op('BUILD_LIST', 103)       # Number of list items
 def_op('BUILD_SET', 104)        # Number of set items
-def_op('BUILD_MAP', 105)        # Number of dict entries (upto 255)
+def_op('BUILD_MAP', 105)        # Number of dict entries (up to 255)
```
Comment (serhiy-storchaka) [inline Lib/opcode.py]: "(up to 255)" is outdated (in 3.6 too). Just remove it.

File: Lib/unittest/test/test_discovery.py [hunk 7]
```diff
@@ -199,8 +199,8 @@ def loadTestsFromModule(module, pattern=None):
                          ['a_directory', 'test_directory', 'test_directory2'])
 
         # load_tests should have been called once with loader, tests and pattern
-        # (but there are no tests in our stub module itself, so thats [] at the
-        # time of call.
+        # (but there are no tests in our stub module itself, so that is [] at
+        # the time of call.
```
Comment (serhiy-storchaka) [inline Lib/unittest/test/test_discovery.py]: Close parenthesis.

File: aclocal.m4 [hunk 8]
```diff
@@ -100,7 +100,7 @@ dnl Check to see whether a particular set of modules exists. Similar to
 dnl PKG_CHECK_MODULES(), but does not set variables or print errors.
 dnl
 dnl Please remember that m4 expands AC_REQUIRE([PKG_PROG_PKG_CONFIG])
-dnl only at the first occurence in configure.ac, so if the first place
+dnl only at the first occurrence in configure.ac, so if the first place
```
Comment (serhiy-storchaka) [inline aclocal.m4]: Shouldn't `configure` be regenerated now?

Comment (terryjreedy) [reply aclocal.m4]: I don't know and could not do so.  Perhaps this should be reverted and either done separately, for 3.8 only, or left alone.

File: Lib/opcode.py (hunk 6 above)
Comment (terryjreedy) [reply Lib/opcode.py]: elopio or whoever makes the change should refresh the patch against master.

File: aclocal.m4 (hunk 8 above)
Comment (terryjreedy) [reply aclocal.m4]: All 3 changes made.  Anyone who wants to fix 'occurence' in aclocal.m4 for 3.8 and regenerate `configure` or whatever should do so separately.

**Comment by elopio** [comment]
thanks everybody!!! *hugs*

**Pull request merged**